pkg/auth/user/util.go

```diff
@@ -25,3 +27,14 @@ func ConvertAttributes(attrMap map[string][]string) []*v1.UserAttribute {
 	})
 	return result
 }
+
+// LogSuccessfulUserLogin logs user attributes in the specified logger instance.
+func LogSuccessfulUserLogin(log *logging.Logger, user *v1.AuthStatus) {
+	loggableUser := user.Clone()
+	// Auth provider config can contain sensitive data(client secret, certificates etc.) so it shouldn't be logged.
+	if loggableUser != nil {
+		loggableUser.AuthProvider = nil
```

Comment:
Do you know if our logging framework supports obfuscating secrets? If we set the auth provider to nil, it could lead to wrong conclusions when viewing the logs, and we also lose all non-secret data.

Reply:
I don't see `zap` supporting secret obfuscating out of the box. We could do this by introducing a custom type for a secret field, but unfortunately, our client secrets are kept in `map[string]string` structure.

Overall, the idea behind my decision is two-fold: the data contained within auth provider config can be sensitive + not needed. If you want to determine which auth provider was used to log in the user, perhaps we can log just auth provider id?

Reply:
I do think that obfuscating sensitive values leads to more insightful logs and is therefore superior to just setting them to `nil`. How about creating a ticket to add such a functionality? Until then we can go with the `nil` solution to unblock the login logs.

Reply:
I'll create a ticket

Reply:
Why would we not be reusing the `[RedactConfig](https://github.com/stackrox/stackrox/blob/21d0873bacc91ae9b27771f09d655e0c8c66e6a9/pkg/auth/authproviders/backend_factory.go#L37)` method from each provider instead of recreating some different way? This is already used to publish config to the UI, so it'd be a good opportunity to use it here as well.

Reply:
I switched to logging selected fields instead of the entire `v1.AuthStatus`, which should solve this issue.